enrichments/internal/elastic/log.go

```diff
@@ -20,15 +20,14 @@ package elastic
 import (
 	"github.com/elastic/opentelemetry-lib/elasticattr"
 	"github.com/elastic/opentelemetry-lib/enrichments/config"
+	"github.com/elastic/opentelemetry-lib/enrichments/internal/attribute"
 	"github.com/elastic/opentelemetry-lib/enrichments/internal/elastic/mobile"
 	"go.opentelemetry.io/collector/pdata/plog"
 )
 
 func EnrichLog(resourceAttrs map[string]any, log plog.LogRecord, cfg config.Config) {
-	if cfg.Log.ProcessorEvent.Enabled {
-		if _, exists := log.Attributes().Get(elasticattr.ProcessorEvent); !exists {
-			log.Attributes().PutStr(elasticattr.ProcessorEvent, "log")
-		}
+	if cfg.Log.ProcessorEvent.Enabled && attribute.IsEmpty(log.Attributes(), elasticattr.ProcessorEvent) {
```

Comment:
The previous logic checked for existence of the attribute key, under what cases will we have a key without any values set?

Reply:
The only case where this can happen is if an empty string is explicitly for the `processor.event`. I added `attribute.IsEmpty` to have a consistent way to check if an attr exists/is empty but I think it also helps cover this edge case where an empty attr value is set.